File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int count = 0;
        if (arr.length < k) {
            return count;
        }
        int sum = 0;
        for (int i = 0; i < k; i++) {
            sum = sum + arr[i];
        }
        if (sum * 1.0 / k >= threshold) {
            count++;
        }
        for (int i = k; i < arr.length; i++) {
            sum = sum - arr[i - k];
            sum = sum + arr[i];
            if (sum * 1.0 / k >= threshold) {
                count++;
            }
        }
        return count;
    }
}

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int n = arr.size();
        int left = 0;
        int right = 0;
        int sum = 0;
        int count = 0;
        while (right < n) {
            sum += arr[right];
            right++;
            while (right - left >= k && sum >= threshold * k) {
                if (right - left == k) {
                    count++;
                }
                sum -= arr[left];
                left++;
            }
        }
        return count;
    }
};